python/equistore-operations/equistore/operations/manipulate_dimension.py

```diff
@@ -210,8 +216,12 @@ def remove_dimension(tensor: TensorMap, axis: str, name: str) -> TensorMap:
         properties = block.properties
 
         if axis == "samples":
+            if len(np.unique(samples.column(name))) > 1:
```

Comment:
I don't think this check works: we could have a column with `-1, -1, -2, -2` but when removing it the values are still unique

Reply:
Hmm yeah good point. It seems that @PicoCentauri also is doing a similar check in `join` https://github.com/lab-cosmo/equistore/pull/301. Perhaps it would be good to have a utility function `def column_is_redundant(labels, name) -> bool:` that does the proper check and can be used both here and in `join`?

Reply:
Thoughts @Luthaf @PicoCentauri ?

Reply:
Function could be useful, but I don't know if it should be public.

Reply:
What's wrong with trying to remove it and then letting the exception bubble up? Potentially catching the root exception and throwing a new one with more context

Reply:
Arf, I forgot about that ... I don't see a much better way than making the rust-side error clearer then =/

Reply:
Ok yeah I think that's a good alternative :)

Reply:
Do you want to do it @jwa7 ?

If yes maybe #306 can help to find the rust files you want to look at.

Reply:
The error comes from here: https://github.com/lab-cosmo/equistore/blob/139958001633368b1ffd725aca916084df724d29/equistore-core/src/labels.rs#L172-L174

We are validating entry by entry the new Labels in a LabelsBuilder, while computing and cache the position of each entry. If is it helpful, we could move the validation in another function.

Reply:
After a discussion with @PicoCentauri it seems that it is not so easy to modify the error message rust-side in a helpful way. We are going back to the idea of a separate function that checks for redundancy, maybe as a private function. I think this would just be a near-copy of the code in Philip's `join` PR, but generalised of course. THoughts @Luthaf ?